src/FakeItEasy/Core/DefaultFakeManagerAccessor.cs

```diff
@@ -26,7 +26,21 @@ public FakeManager GetFakeManager(object proxy)
 
             if (result == null)
             {
-                throw new ArgumentException("The specified object is not recognized as a fake object.");
+                object tag = null;
+                ITaggable taggable = proxy as ITaggable;
+                if (taggable != null && taggable.Tag != null)
+                {
+                    tag = taggable.Tag;
+                }
+
+                if (tag == null)
+                {
+                    throw new ArgumentException(string.Format(ExceptionMessages.NotRecognisedAsFake, proxy.GetType().FullName));
+                }
+                else
+                {
+                    throw new ArgumentException(string.Format(ExceptionMessages.NotRecognisedAsFake, tag.GetType().FullName));

```

Comment:
Alright sounds good, thanks guys. What do you suggest doing in the case of that one unit test which hits the alternate logic branch?